kernel/comps/block/src/bio.rs

```diff
@@ -406,25 +405,39 @@ impl<'a> BioSegment {
         self.len.value()
     }
 
+    /// Returns the offset (in bytes) within the first page.
+    pub fn offset(&self) -> usize {
+        self.offset.value()
+    }
+
+    /// Returns the contiguous pages on which this segment resides.
+    pub fn pages(&self) -> &VmSegment {
+        &self.pages
+    }
+
     /// Returns a reader to read data from it.
     pub fn reader(&'a self) -> VmReader<'a> {
-        let reader = match &self.pages {
-            Pages::Segment(segment) => segment.reader(),
-            Pages::Frame(frame) => frame.reader(),
-        };
-        reader.skip(self.offset.value()).limit(self.len.value())
+        self.pages
+            .reader()
+            .skip(self.offset.value())
+            .limit(self.len.value())
     }
 
     /// Returns a writer to write data into it.
     pub fn writer(&'a self) -> VmWriter<'a> {
-        let writer = match &self.pages {
-            Pages::Segment(segment) => segment.writer(),
-            Pages::Frame(frame) => frame.writer(),
-        };
-        writer.skip(self.offset.value()).limit(self.len.value())
+        self.pages
+            .writer()
+            .skip(self.offset.value())
+            .limit(self.len.value())
     }
 }
 
+fn frame_range(byte_range: &Range<usize>) -> Range<usize> {
```

Comment:
I think the changes to `Bio` should be separated from the support of conversion from `VmFrame` to `VmSegment` by adding another commit. Please write more atomic commits next time.